apps/docs-snippets/src/guide/transactions/transaction-request.test.ts

```diff
@@ -0,0 +1,191 @@
+    DocSnippetProjectsEnum.SUM_SCRIPT
+  );
+
+  const { abiContents: predicateAbi, binHexlified: predicateBytecode } = getDocsSnippetsForcProject(
+    DocSnippetProjectsEnum.SIMPLE_PREDICATE
+  );
+
+  const address = Address.fromRandom();
+
+  const message = {
+    assetId: BaseAssetId,
+    sender: address,
+    recipient: address,
+    nonce: '0x',
+    amount: bn(0),
+    daHeight: bn(0),
+  };
+  const coin: Coin = {
+    id: '0x',
+    assetId: BaseAssetId,
+    amount: bn(0),
+    owner: address,
+    maturity: 1,
+    blockCreated: bn(0),
+    txCreatedIdx: bn(0),
+  };
+
+  beforeAll(async () => {
+    provider = await Provider.create(FUEL_NETWORK_URL);
+  });
+
+  it('creates a transaction request from ScriptTransactionRequest', () => {
+    const scriptMainFunctionArguments = [1];
+
+    // #region transaction-request-1
+    // #import { ScriptTransactionRequest };
+
+    // Instantiate the transaction request using a ScriptTransactionRequest
+    const transactionRequest = new ScriptTransactionRequest({ script: scriptBytecode });
+
+    // Set the script main function arguments (can also be passed in the class constructor)
+    transactionRequest.setData(scriptAbi, scriptMainFunctionArguments);
+    // #endregion transaction-request-1
+
+    expect(transactionRequest.script).toEqual(arrayify(scriptBytecode));
+  });
+
+  it('creates a transaction request fromm a CreateTransactionRequest', () => {
+    const contractByteCode = ZeroBytes32;
+
+    // #region transaction-request-2
+    // #import { CreateTransactionRequest };
+
+    // Instantiate the transaction request using a CreateTransactionRequest
+    const transactionRequest = new CreateTransactionRequest({ witnesses: [contractByteCode] });
+    // #endregion transaction-request-2
+
+    expect(transactionRequest.witnesses[0]).toEqual(contractByteCode);
+  });
+
+  it('modifies a transaction request', () => {
+    const recipientAddress = address;
+    const resource = coin;
+    const resources: Resource[] = [resource];
+
+    // #region transaction-request-3
+    // #import { ScriptTransactionRequest, BaseAssetId };
+
+    // Instantiate the transaction request
+    const transactionRequest = new ScriptTransactionRequest({ script: scriptBytecode });
+
+    // Adding resources (coins or messages)
+    transactionRequest.addResources(resources);
+    transactionRequest.addResource(resource);
+
+    // Adding coin inputs and outputs (including transfer to recipient)
+    transactionRequest.addCoinInput(coin);
+    transactionRequest.addCoinOutput(recipientAddress, 1000, BaseAssetId);
+
+    // Adding message inputs
+    transactionRequest.addMessageInput(message);
+    // #endregion transaction-request-3
+
+    expect(transactionRequest.script).toEqual(arrayify(scriptBytecode));
+    expect(transactionRequest.inputs.length).toEqual(4);
+    expect(transactionRequest.outputs.length).toEqual(2);
+    expect(transactionRequest.witnesses.length).toEqual(1);
```

Comment:
The flow should be:
```ts
    const transactionRequest = new ScriptTransactionRequest({ script: scriptBytecode });

    transactionRequest.addCoinOutput(recipientAddress, 1000, BaseAssetId);

    // Adding message inputs
    // Is there a reason why a message is being added? If so it needs to be added before estimating the TX
    transactionRequest.addMessageInput(message);

    const { gasUsed, requiredQuantities, maxFee } = provider.getTransactionCost(transactionRequest);

    transactionRequest.gasLimit = gasUsed;

     // Adding resources (coins or messages)
    await senderWallet.fund(transactionRequest, requiredQuantities, maxFee);
    // #endregion transaction-request-3
```

The reason is that we should avoid adding resources without knowing the real amount the TX will require ( in this specific case it's the transfer and the fee ) and avoid running `OutOfGas`.

Reply:
Hmm this snippet is attempting to demo methods available to a tx request to modify it, rather than flow. Hence the more verbose approach and transaction submission is not included. And we are only testing the shape of the tx. Do you still think we need to ensure the flow is correct?